mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -0,0 +1,165 @@
+/// format, it decrypts it before returning.
+///
+/// # Arguments
+/// * `ctx` - The `MmArc` context containing the application state and configuration.
+/// * `wallet_name` - The name of the wallet for which the passphrase is to be retrieved.
+///
+/// # Returns
+/// `MmInitResult<String>` - The decrypted passphrase or an error if any operation fails.
+///
+/// # Errors
+/// Returns specific `MmInitError` variants for different failure scenarios.
+async fn read_and_decrypt_passphrase(
+    ctx: &MmArc,
+    wallet_name: &str,
+    wallet_password: &str,
+) -> WalletInitResult<Option<String>> {
+    match read_encrypted_passphrase(ctx, wallet_name)
+        .await
+        .mm_err(|e| WalletInitError::WalletsStorageError(e.to_string()))?
+    {
+        Some(encrypted_passphrase) => {
+            let mnemonic = decrypt_mnemonic(&encrypted_passphrase, wallet_password)?;
+            Ok(Some(mnemonic.to_string()))
+        },
+        None => Ok(None),
+    }
+}
+
+/// Initializes and manages the wallet passphrase.
+///
+/// This function handles several scenarios based on the configuration:
+/// - Deserializes the passphrase and wallet name from the configuration.
+/// - If both wallet name and passphrase are `None`, the function sets up the context for "no login mode"
+///   This mode can be entered after the function's execution, allowing access to Komodo DeFi Framework
+///   functionalities that don't require a passphrase (e.g., viewing the orderbook).
+/// - If a wallet name is provided without a passphrase, it first checks for the existence of a
+///   passphrase file associated with the wallet. If no file is found, it generates a new passphrase,
+///   encrypts it, and saves it, enabling multi-wallet support.
+/// - If a passphrase is provided (with or without a wallet name), it uses the provided passphrase
+///   and handles encryption and storage as needed.
+/// - Initializes the cryptographic context based on the `enable_hd` configuration.
+///
+/// # Arguments
+/// * `ctx` - The `MmArc` context containing the application state and configuration.
+///
+/// # Returns
+/// `MmInitResult<()>` - Result indicating success or failure of the initialization process.
+///
+/// # Errors
+/// Returns `MmInitError` if deserialization fails or if there are issues in passphrase handling.
+///
+pub(crate) async fn initialize_wallet_passphrase(ctx: MmArc) -> WalletInitResult<()> {
+    let passphrase = deserialize_config_field::<Option<String>>(&ctx, "passphrase")?;
+    // New approach for passphrase, `wallet_name` is needed in the config to enable multi-wallet support.
+    // In this case the passphrase will be generated if not provided.
+    // The passphrase will then be encrypted and saved whether it was generated or provided.
+    let wallet_name = deserialize_config_field::<Option<String>>(&ctx, "wallet_name")?;
+
+    let passphrase = match (wallet_name, passphrase) {
```

Comment:
Relying on existence of `wallet_name` might a bit confusing (just IMHO/nitpick). What do you think about using more explicit config field like "use_encrypted_wallet"?

Reply:
I think it's not needed actually :slightly_smiling_face: Thanks for the detailed explanation!

Reply:
Yeah, I guess I shouldn't use a struct for backward compatibility of the `passphrase` field. I will add the option to pass an encrypted passphrase in next iteration, so I will leave this unresolved until then.

Reply:
> I will add the option to pass an encrypted passphrase for the case of uploading a seed file in next iteration, so I will leave this unresolved until then.

Done